typescript/infra/scripts/verify-validator.ts

```diff
@@ -0,0 +1,300 @@
+      break;
+    }
+
+    const key = `checkpoint_${i}.json`;
+
+    let c: Checkpoint | null;
+    try {
+      const t = await cClient.getS3Obj(key);
+      if (isCheckpoint(t)) {
+        if (t.checkpoint.index != i) {
+          console.log(`${i}: Control index is invalid`, t);
+          process.exit(1);
+        }
+        c = t;
+      } else {
+        console.log(`${i}: Invalid control checkpoint`, t);
+        process.exit(1);
+      }
+    } catch (err) {
+      c = null;
+    }
+
+    let p: Checkpoint;
+    try {
+      const t = await pClient.getS3Obj(key);
+      if (isCheckpoint(t)) {
+        p = t;
+      } else {
+        console.log(`${i}: Invalid prospective checkpoint`, t);
+        invalidCheckpoints.push(i);
+        continue;
+      }
+      if (!c) {
+        extraCheckpoints.push(i);
+      }
+      missingInARow = 0;
+    } catch (err) {
+      missingCheckpoints.push(i);
+      missingInARow++;
+      continue;
+    }
+
+    console.assert(
+      p.checkpoint.index == i,
+      `${i}: checkpoint indexes do not match`,
+    );
+
+    // TODO: verify signature
+
+    if (!c) {
+      continue;
+    }
+
+    // compare against the control
+    console.assert(
+      c.checkpoint.outbox_domain == p.checkpoint.outbox_domain,
+      `${i}: outbox_domains do not match`,
+    );
+    console.assert(
+      c.checkpoint.root == p.checkpoint.root,
+      `${i}: checkpoint roots do not match`,
+    );
+
+    try {
+      const [cLastMod, pLastMod] = await Promise.all([
+        cClient.getLastModTime(key),
+        pClient.getLastModTime(key),
+      ]);
+      const diffS = (pLastMod.valueOf() - cLastMod.valueOf()) / 1000;
+      if (Math.abs(diffS) > 10) {
+        console.log(`${i}: Modification times differ by ${diffS}s`);
+      }
+      modTimeDeltasS.push(diffS);
+    } catch (err) {
+      // this is probably a permission error since we already know they should exist
+      console.error(`${i}: Error validating last modified times`, err);
+    }
+
+    fullyCorrectCheckpoints.push(i);
+  }
+
+  if (fullyCorrectCheckpoints.length)
+    console.log(
+      `Fully correct checkpoints (${fullyCorrectCheckpoints.length}): ${fullyCorrectCheckpoints}\n`,
+    );
+  if (extraCheckpoints.length)
+    console.log(
+      `Extra checkpoints (${extraCheckpoints.length}): ${extraCheckpoints}\n`,
+    );
+  if (missingCheckpoints.length)
+    console.log(
+      `Missing checkpoints (${missingCheckpoints.length}): ${missingCheckpoints}\n`,
+    );
+  if (invalidCheckpoints.length)
+    console.log(
+      `Invalid checkpoints (${invalidCheckpoints.length}): ${invalidCheckpoints}\n`,
+    );
+
+  if (modTimeDeltasS.length) {
+    console.log(modTimeDeltasS);
+    console.log(
+      `Time deltas (∆< 0 -> prospective came earlier than the control)`,
+    );
+    console.log(`Median: ${median(modTimeDeltasS)}s`);
+    console.log(`Mean:   ${mean(modTimeDeltasS)}s`);
+    console.log(`Stdev:  ${stdDev(modTimeDeltasS)}s`);
```

Comment:
Noticing that the mean / std deviation are pretty skewed by the timestamp of the first signature the prospective validator does upon startup. Not sure if I have a great suggestion apart from maybe ignoring the first checkpoint signed by the prospective one, which feels a _little_ weird but may be worth it to make the mean & std deviation make sense. E.g. when running `ts-node ./scripts/verify-validator.ts -a 0x000 -c https://abacus-mainnet-celo-validator-0.s3.us-east-1.amazonaws.com/ -p https://abacus-mainnet-zkv-celo-signatures.s3.eu-west-2.amazonaws.com/`:

```
[
  -41,  3,  3,  3,  2,  2,  4,  5, -1,  0,     0,
    0, -1,  0,  3,  3,  3, -1, -4, -3, -3,    -3,
   -3,  1,  1,  0,  0,  0,  1,  1,  0,  1,     0,
    0,  0,  0,  0, -2, -3, -3, -2, -2,  0,    -1,
    0, -1, -1, -1, -1, -1, -2, -3, -3, -3,    -4,
   -1, -2, -2, -2, -1,  3,  4,  4,  4,  3,     9,
   10,  4,  4,  8,  9,  9,  9,  8,  6,  6,     6,
    6,  5,  4,  5,  5,  5,  5,  5,  5,  4,     5,
    5,  7,  7,  7,  7,  8,  9, 10,  5,  4, 82566
]
Time deltas (∆< 0 -> prospective came earlier than the control)
Median: 1s
Mean:   835.6565656565657s
Stdev:  8298.197339838882s
```

Reply:
Yeah, I think throwing out that first one makes sense